### DISCUSSION TITLE
Is there option to increase the players’ buffer size?

### DISCUSSION THREAD

**Discussion by holefrog** [sle118/squeezelite-esp32#379] [Q&A]
Actually, I have two questions.

1. **Audio Capability Issue:**
I'm dealing with a FLAC file featuring a sample rate of 44.1kHz, 16-bit depth, and a bitrate of 1039kbps. While it usually plays without any problem, at times, it becomes intermittent. During these occurrences, the "Buffer Fullness" drops rapidly from 100% to 0.

2. **Buffer Size Adjustment:**
Is there a way to increase the playerm‘s buffer size? I don't mind if it takes several seconds to buffer, as long as it helps resolve the intermittent playback issue.

**Comment by Renber78** [comment]
You can adjust the "-b" parameter of the squeezelite command line. It is factory set to 500:2000 kbytes
See the squeezelite documentation for more info.

**Reply by holefrog** [reply]
Thank you Renber78,

The default of "Stream and Output buffer sizes (in Kbytes)"   is 500:2000. 

I have no idea which are they meaning.  
Stream buffer: the buffer of data received from LMS?
Output buffer: the buffer of data sent to DAC?

Which buffer does the "Buffer Fullness" indicate? 

I now set them to: 525:2100.  Are the bigger the better? 

The playback is still paused, showing 'Rebuffering' from 0% to 100%. However, the 'Buffer Fullness' remains at 0% even after the rebuffering reaches 100%.

Furthermore, I also encountered the 'stream_ogg:200 OggS not at expected position' error after modifying the buffers. Additionally, using Telnet may lead to a reboot at first connecting attempt.

**Reply by sle118** [reply]
You didn't specify your setup. Are you using i2s or playing back to an external Bluetooth speaker? This makes a huge difference, as the radio time would need to be split between wifi and Bluetooth.

If you are outputting to i2s, then I suggest you try moving the player closer to the wifi access point to see if that makes a difference. You could also try a different channel for your access point if you think the channel could be too busy.

Wifi is very sensitive to noise, especially to microwave ovens because these operate in a similar frequency range as wifi. Increasing buffer may improve stability, but the issue seems to be caused by an inefficient data transfer

**Reply by jack338c** [reply]
if use external Bluetooth speaker,then how to set  "Stream and Output buffer sizes (in Kbytes)" ?
Could you share some tips,thanks

**Comment by holefrog** [comment]
Thank you so much for the reply.

> You didn't specify your setup. Are you using i2s or playing back to an external Bluetooth speaker? This makes a huge difference, as the radio time would need to be split between wifi and Bluetooth.

I play songs from LMS via WIFI, using a 'Louder ESP' with a Stereo I2S DAC.  (When it comes to Bluetooth playback, I have a very serious problem.)
Here is my config:
```
{
  "a2dp_ctmt": "1000",
  "a2dp_ctrld": "500",
  "a2dp_dev_name": "ESP32-Louder",
  "a2dp_sink_name": "SMSL BT4.2",
  "a2dp_spin": "0000",
  "actrls_config": "",
  "airplay_name": "ESP32-Louder",
  "airplay_port": "5000",
  "ap_channel": "1",
  "ap_ip_address": "192.168.4.1",
  "ap_ip_gateway": "192.168.4.1",
  "ap_ip_netmask": "255.255.255.0",
  "ap_pwd": "squeezelite",
  "ap_ssid": "ESP32-Louder",
  "autoexec": "1",
  "autoexec1": "squeezelite  -o i2s  -s  192.168.50.210:3483 -b  525:2100 -C  30 -d  all=info ",
  "bat_config": "",
  "bt_name": "ESP32-Louder",
  "bt_sink_pin": "1234",
  "bt_sink_volume": "127",
  "bypass_wm": "0",
  "cspot_config": "{\t\"deviceName\":\t\"ESP32-Louder\",\t\"bitrate\":\t160,\t\"zeroConf\":\t1}",
  "dac_config": "model=I2S,bck=26,ws=25,do=22,sda=21,scl=27,i2c=45 ",
  "dac_controlset": "{\"init\":[{\"reg\":3,\"val\":2},{\"reg\":3,\"val\":3}],\"poweron\":[{\"reg\":3,\"val\":3}],\"poweroff\":[{\"reg\":3,\"val\":0}]}",
  "dhcp_tmout": "8",
  "display_config": "SPI,width=256,height=64,reset=23,cs=5,speed=8000000,driver=SSD1322",
  "enable_airplay": "Y",
  "enable_bt_sink": "Y",
  "enable_cspot": "N",
  "equalizer": "0,0,0,0,0,0,0,0,0,0",
  "eth_config": "",
  "ethtmout": "8",
  "gpio_exp_config": "",
  "host_name": "ESP32-Louder",
  "i2c_config": "scl=27,sda=21,speed=400000,port=1 ",
  "jack_mutes_amp": "n",
  "led_brightness": "",
  "led_vu_config": "",
  "lms_ctrls_raw": "n",
  "loudness": "0",
  "metadata_config": "",
  "model_config": "",
  "ota_erase_blk": "249856",
  "ota_prio": "6",
  "ota_stack": "10240",
  "pollmin": "15",
  "pollmx": "600",
  "rel_api": "https://api.github.com/repos/sle118/squeezelite-esp32/releases",
  "release_url": "CONFIG_SQUEEZELITE_ESP32_RELEASE_URL",
  "rotary_config": "A=13, B=14, SW=35, volume, longpress",
  "set_GPIO": "33=vcc, 36=ir",
  "sleep_config": "",
  "spdif_config": "",
  "spi_config": "data=19,clk=18,dc=32",
  "stats": "N",
  "target": "",
  "telnet_block": "500",
  "telnet_buffer": "40000",
  "telnet_enable": "D"
}
```

> 
> If you are outputting to i2s, then I suggest you try moving the player closer to the wifi access point to see if that makes a difference. You could also try a different channel for your access point if you think the channel could be too busy.

The player is several meters away from my router; according to the settings of LMS, the 'Wireless Signal Strength' is 89%. I used the WiFi Analyzer app to find the best channel and, according to the app, my WiFi signal strength is twice that of others.

> 
> Wifi is very sensitive to noise, especially to microwave ovens because these operate in a similar frequency range as wifi. Increasing buffer may improve stability, but the issue seems to be caused by an inefficient data transfer

You are absolutely right about the microwave oven; I noticed it got much worse when the oven was running. I thought it might be because of the interference too. Therefore, I want to increase the input buffer to make it more stable. What is your suggestion for the buffer?
